march_hardware/src/Joint.cpp

```diff
@@ -23,30 +24,24 @@ void Joint::initialize(int ecatCycleTime)
 
 void Joint::prepareActuation()
 {
-  ROS_ASSERT_MSG(this->getActuationMode() != ActuationMode::unknown, "The mode of actuation for joint %s is: %s",
-                 this->name.c_str(), this->getActuationMode().toString().c_str());
-  if (this->allowActuation)
+  if (!this->allowActuation)
   {
-    ROS_INFO("[%s] Preparing for actuation", this->name.c_str());
-    this->iMotionCube.goToOperationEnabled();
-    ROS_INFO("[%s] Successfully prepared for actuation", this->name.c_str());
-  }
-  else
-  {
-    ROS_ERROR("[%s] Trying to prepare for actuation while it is not "
-              "allowed to actuate",
-              this->name.c_str());
+    throw error::HardwareException(error::ErrorType::NOT_ALLOWED_TO_ACTUATE, "Failed to prepare joint %s for actuation",
+                                   this->name.c_str());
   }
+  ROS_INFO("[%s] Preparing for actuation", this->name.c_str());
+  this->iMotionCube.goToOperationEnabled();
+  ROS_INFO("[%s] Successfully prepared for actuation", this->name.c_str());
 }
 
 void Joint::actuateRad(double targetPositionRad)
 {
-  ROS_ASSERT_MSG(this->allowActuation,
-                 "Joint %s is not allowed to actuate, "
-                 "yet its actuate method has been called",
-                 this->name.c_str());
-  // TODO(BaCo) check that the position is allowed and does not exceed (torque)
-  // limits.
+  if (!this->allowActuation)
+  {
+    throw error::HardwareException(error::ErrorType::NOT_ALLOWED_TO_ACTUATE, "Joint %s is not allowed to actuate",
+                                   this->name.c_str());
+  }
+  // TODO(BaCo) check that the position is allowed and does not exceed (torque) limits.
```

Comment:
Should we make an issue for this?

Reply:
Maybe we should check all the `TODO`s in the code and create issues from them if necessary. I'll do it right now